packages/eslint-plugin/tests/rules/prefer-optional-chain/prefer-optional-chain.test.ts

```diff
@@ -665,35 +665,393 @@ describe('|| {}', () => {
+            line */
+            // single-line
+            foo.bar(); /* after semicolon */
+            // after expression
+          }
+        `,
+        errors: [{ messageId: 'preferOptionalChain' }],
+        options: [{ ignoreIfStatements: false }],
+        output: `
+          declare const foo: undefined | { bar: () => void };
+          // before expression
+            /** multi
+            line */
+            // single-line
+          foo?.bar();
+          /* after semicolon */
+            // after expression
+        `,
+      },
+      {
+        // eslint-disable-next-line @typescript-eslint/internal/plugin-test-formatting
+        code: `
+          declare const foo: undefined | { bar: () => void };
+          if /* sneaky1 */ (/* sneaky2 */ foo /* sneaky3 */) /* sneaky4 */ {
+            // comment1
+            /* comment2 */
+            foo.bar();
+            // comment3
+            /* comment4 */
+          }
+        `,
+        errors: [
+          {
+            messageId: 'preferOptionalChain',
+            suggestions: [
+              {
+                messageId: 'optionalChainSuggest',
+                output: `
+          declare const foo: undefined | { bar: () => void };
+          /* sneaky1 */
+          /* sneaky2 */
+          /* sneaky3 */
+          /* sneaky4 */
+          // comment1
+          /* comment2 */
+          foo?.bar();
+          // comment3
+          /* comment4 */
+        `,
+              },
+            ],
+          },
+        ],
+        options: [{ ignoreIfStatements: false }],
+      },
+
+      {
+        // eslint-disable-next-line @typescript-eslint/internal/plugin-test-formatting
+        code: `
+          declare const foo: undefined | { bar: () => void };
+          if (foo) /* sneaky */
+            foo.bar();
+        `,
+        errors: [{ messageId: 'preferOptionalChain' }],
+        options: [{ ignoreIfStatements: false }],
+        output: `
+          declare const foo: undefined | { bar: () => void };
+          /* sneaky */
+          foo?.bar();
+        `,
+      },
+      {
+        code: `
+          declare const foo: undefined | { bar: () => void };
+          if (foo) {
+            foo.bar(); // comment
+          }
+        `,
+        errors: [{ messageId: 'preferOptionalChain' }],
+        options: [{ ignoreIfStatements: false }],
+        output: `
+          declare const foo: undefined | { bar: () => void };
+          foo?.bar();
+          // comment
+        `,
+      },
+      {
+        code: `
+          declare const foo: undefined | { bar: () => void };
+          if (foo) {
+            foo.bar();
+            // comment 1
+            // comment 2
+          }
+        `,
+        errors: [{ messageId: 'preferOptionalChain' }],
+        options: [{ ignoreIfStatements: false }],
+        output: `
+          declare const foo: undefined | { bar: () => void };
+          foo?.bar();
+          // comment 1
+            // comment 2
```

Comment:
I took the suggestion from here: https://github.com/typescript-eslint/typescript-eslint/pull/10137#discussion_r1854400776
This preserves the range between the first and last comments, so the indent before the first comment is modified.

I feel like it's impossible to get 100% correctness when moving comments. Do we want to change the indent of all of them?

Reply:
We don't care deeply about indentation or spacing in general, that's a formatter's concern. As long as the comment is in a reasonably similar location that seems to match up we're happy.

If we can do some very basic work to make the comments a little prettier that's fine. But we don't want to add a lot of formatting-specific code that makes the file harder to read.